### PR TITLE
batches: make `-f` optional

### PR DESCRIPTION
**Pull request by LawnGnome**
New behaviour (assuming `COMMAND` is any command that accepted `-f`, including `apply`, `preview`, `repos`, or `validate`):

* `src batch COMMAND` **on a TTY**: will wait for 250ms in case there's input on stdin; otherwise errors
* `src batch COMMAND` with a file redirected in: waits indefinitely for input
* `src batch COMMAND -f -`: waits indefinitely for input
* `src batch COMMAND FILE`: reads from `FILE`; equivalent to `src batch COMMAND -f FILE`
* `src batch COMMAND -f FILE`: works as before

In summary, this shouldn't break backward compatibility in any case except for the user _slowly_ typing in a batch spec on stdin.

Fixes #680.

### CODE REVIEW

File: internal/batches/ui/tty.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package ui
+
+import (
+	"time"
+
+	"github.com/creack/goselect"
+)
+
+func HasInput(fd uintptr, timeout time.Duration) (bool, error) {
+
```
Comment (eseliger) [inline internal/batches/ui/tty.go:10]:
```suggestion
```

File: cmd/src/batch_common.go [hunk 2]
```diff
@@ -176,14 +194,31 @@ func batchDefaultTempDirPrefix() string {
 	return os.TempDir()
 }
 
-func batchOpenFileFlag(flag *string) (io.ReadCloser, error) {
-	if flag == nil || *flag == "" || *flag == "-" {
+func batchOpenFileFlag(flag string) (io.ReadCloser, error) {
```
Comment (eseliger) [inline cmd/src/batch_common.go:197]: this is also used in the exec path - not sure we want this there? anyways, not super important.

**Review by eseliger — APPROVED** [review]

File: cmd/src/batch_common.go (hunk 2 above)
Comment (LawnGnome) [reply cmd/src/batch_common.go:197]: It's not particularly relevant, but I also don't think it'll cause problems. We can pull out the open/validate bits we need for `exec` at a later date if we want. (I don't mind doing that, but I also don't think it's on the critical path here.)

Comment (eseliger) [reply cmd/src/batch_common.go:197]: yeah no, let's leave it as is.
